### ISSUE TITLE
Crash when streaming over Airplay 2

### ISSUE THREAD

**Issue by badrabbit** [jishi/node-sonos-http-api#633]
Every time i start to play music from my iPhone (iOS 12 beta) to an Airplay-2-enabled SONOS Speaker or Group, the API crashes:

```
macmini-homes-Mac-mini:SONOS_HTTP_API macmini_home$ node server.js
2018-07-23T13:49:11.306Z INFO Could not find file /Users/macmini_home/SONOS_HTTP_API/settings.json
2018-07-23T13:49:11.491Z INFO Presets loaded: {}
2018-07-23T13:49:11.669Z INFO http server listening on port 5005
/Users/macmini_home/SONOS_HTTP_API/node_modules/sonos-discovery/lib/models/Player.js:99
      track.duration = parseTime((item.res.$attrs || item.res).duration);
                                           ^

TypeError: Cannot read property '$attrs' of undefined
    at EventEmitter.sax.on (/Users/macmini_home/SONOS_HTTP_API/node_modules/sonos-discovery/lib/models/Player.js:99:44)
    at EventEmitter.emit (events.js:182:13)
    at EventEmitter.emit (domain.js:442:20)
    at SAXStream.onCloseTag (/Users/macmini_home/SONOS_HTTP_API/node_modules/xml-flow/lib/xml-flow.js:131:15)
    at SAXStream.emit (events.js:182:13)
    at SAXStream.EventEmitter.emit (domain.js:442:20)
    at SAXParser.me._parser.(anonymous function) [as onclosetag] (/Users/macmini_home/SONOS_HTTP_API/node_modules/xml-flow/node_modules/sax/lib/sax.js:258:17)
    at emit (/Users/macmini_home/SONOS_HTTP_API/node_modules/xml-flow/node_modules/sax/lib/sax.js:624:35)
    at emitNode (/Users/macmini_home/SONOS_HTTP_API/node_modules/xml-flow/node_modules/sax/lib/sax.js:629:5)
    at closeTag (/Users/macmini_home/SONOS_HTTP_API/node_modules/xml-flow/node_modules/sax/lib/sax.js:889:7)
    at SAXParser.write (/Users/macmini_home/SONOS_HTTP_API/node_modules/xml-flow/node_modules/sax/lib/sax.js:1436:13)
    at SAXStream.write (/Users/macmini_home/SONOS_HTTP_API/node_modules/xml-flow/node_modules/sax/lib/sax.js:239:18)
    at Readable.ondata (_stream_readable.js:669:20)
    at Readable.emit (events.js:182:13)
    at Readable.EventEmitter.emit (domain.js:442:20)
    at addChunk (_stream_readable.js:277:12)

```

**Comment by jishi** [comment]
Good find. Could you run the API with something like this:

`NODE_LOG_LEVEL=trace node server.js > debug.log`

Make it crash, then send the file `debug.log` to me via email (jimmy at shimizu.se) 

I don't have an iOS device myself nor an Airplay2 compatible player. I just need to see what data the player sends when using Airplay so I can sure I parse it correctly.

**Comment by badrabbit** [comment]
Log is sent.

**Comment by badrabbit** [comment]
Is there something you can do to prevent it from crashing?

**Comment by jishi** [comment]
Yes, it just need more sanity checking, just haven't had time to fix it.

**Comment by jishi** [comment]
Fixed in 1.6.3

**Comment by jishi** [comment]
Remember to update dependencies (`npm install --production`), since this requires 1.7.1 of sonos-discovery!
